plugins/woocommerce-admin/client/products/product-form-actions.tsx

```diff
@@ -73,7 +80,13 @@ export const ProductFormActions: React.FC = () => {
 			...getProductDataForTracks(),
 		} );
 		if ( ! values.id ) {
-			createProductWithStatus( values, 'publish' );
+			const product = await createProductWithStatus( values, 'publish' );
+			if ( product?.id ) {
+				resetForm();
```

Comment:
We should also pass in the `product` into the `resetForm` here.
The transition happens fast enough that the user doesn't notice, but I do get a `uncontrolled component has been changed` warning if we call it without any product object.

Reply:
Ok, but I don't think the right way is passing the product obj here since we are reseting a creation form not an edition one. Which means that when creating the form starts empty. That why I didn't pass the product obj here. If there is a problem regarding to uncontrolled/controlled is because the invalid initial value of the form at the moment it was mounted. Remember the resetForm takes the form's initialValues by default if we don't pass any obj to the function.

Reply:
In that case shouldn't we be resetting it to this object: 
```
{
					reviews_allowed: true,
					name: '',
					sku: '',
					stock_quantity: 0,
					stock_status: 'instock',
				}
```
Which is the same object we initially start the form with.